servers/src/main/java/tachyon/worker/block/BlockMetadataManagerView.java

```diff
@@ -0,0 +1,142 @@
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.StorageTierView;
+
+/**
+ * This class exposes a narrower view of {@link BlockMetadataManager} to Evictors and Allocators.
+ */
+public class BlockMetadataManagerView {
+
+  /** The BlockMetadataManager this view is derived from */
+  private final BlockMetadataManager mMetadataManager;
+  /** A list of StorageTierView, derived from StorageTiers from the BlockMetadataManager */
+  private List<StorageTierView> mTierViews = new ArrayList<StorageTierView>();
+  /** A list of pinned inodes */
+  private final Set<Integer> mPinnedInodes = new HashSet<Integer>();
+  /** A list of blocks that are currently being read */
+  private final Set<Long> mLockedBlocks = new HashSet<Long>();
+
+  /**
+   * Create a BlockMatadataManagerView.
+   * Now we always creating a new view before freespace.
+   * TODO: incrementally update the view
+   *
+   * @param manager
+   * @param pinnedBlocks
+   * @param lockedBlocks
+   * @throws IOException
+   */
+  public BlockMetadataManagerView(BlockMetadataManager manager, Set<Integer> pinnedInodes,
+      Set<Long> lockedBlocks) throws IOException {
+    mMetadataManager = Preconditions.checkNotNull(manager);
+    mPinnedInodes.addAll(Preconditions.checkNotNull(pinnedInodes));
+    mLockedBlocks.addAll(Preconditions.checkNotNull(lockedBlocks));
+
+    // iteratively create all StorageTierViews and StorageDirViews
+    for (StorageTier tier : manager.getTiers()) {
+      mTierViews.add(new StorageTierView(tier, this));
+    }
+  }
+
+  /**
+   * Test if the block is pinned.
+   *
+   */
+  public boolean isBlockPinned(long blockId) {
+    return mPinnedInodes.contains(BlockInfo.computeInodeId(blockId));
+  }
+
+  /**
+   * Test if the block is locked.
+   *
+   */
+  public boolean isBlockLocked(long blockId) {
+    return mLockedBlocks.contains(blockId);
+  }
+
+  /**
+   * Provide StorageTierView given tierAlias
+   *
+   * @param tierAlias the alias of this tierView
+   * @return the StorageTierView object associated with the alias
+   * @throws IOException if tierAlias is not found
+   */
+  public StorageTierView getTierView(int tierAlias) throws IOException {
+    // TODO: can we ensure the returning tierview is same as
+    // new StorageTierView(mMetadataManager.getTier(tierAlias)) ?
+    return mTierViews.get(tierAlias);
+  }
+
+  /**
+   *
+   * @return the list of StorageTierViews
+   */
+  public List<StorageTierView> getTierViews() {
+    return mTierViews;
+  }
+
+  /**
+   *
+   * @param tierAlias the alias of a tier
+   * @return the list of StorageTierView
+   * @throws IOException if tierAlias is not found
+   */
+  public List<StorageTierView> getTierViewsBelow(int tierAlias) throws IOException {
+    // TODO: similar concern as in getTierView
+    int level = getTierView(tierAlias).getTierViewLevel();
+    return mTierViews.subList(level + 1, mTierViews.size());
```

Comment:
Do you mean from `sublist`? I think `a.sublist(a.size(), a.size())` returns a empty list.